common/network-common/src/main/java/org/apache/spark/network/shuffledb/DB.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.shuffledb;
+
+import java.io.Closeable;
+
+/**
+ * The local KV storage used to persist the shuffle state,
+ * the implementations may include LevelDB, RocksDB, etc.
+ */
+public interface DB extends Closeable {
```

Comment:
Is this a developer API or user API? I didn't see the custom DB could be plugged in this PR.

Reply:
developer API.  The implementation of RocksDB is under review: https://github.com/apache/spark/pull/37610,

Reply:
So could you add annotation `@DeveloperApi` for it?

Reply:
OK

Reply:
We do not support users plugging in a different DB instance - it comes from within spark code.
Wont be @DeveloperApi

Reply:
@mridulm It's a public interface. I think it should either be a user API or developer API, no?

Reply:
hmm... should we use `@Private` as `kvstore.KVStore`?

https://github.com/apache/spark/blob/52cd037f321c1e8e1e4d6c1b76b1b4c50fbe3ccd/common/kvstore/src/main/java/org/apache/spark/util/kvstore/KVStore.java#L65-L66

Reply:
The configuration is public @Ngone51, not the implementation itself.
To put it differently, there is no way for users to leverage any of this - unless they modify spark code.

I am fine with marking it as `@Private `if we need to make the intent clearer.

It is the same most submodules in common/* - except probably for common/kvstore, where Marcello marked the interfaces as @Private :-)

Reply:
@mridulm @LuciferYang Thanks.